### PR TITLE
PHPCS: Add `use` statement sorting sniff

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

After a merge of two patches adding `use` statements for the same class in different places in the existing list broke things, discussion on Slack was in favor of adding a sniff to require that `use` statements are in alphabetical order.

As https://github.com/wikimedia/mediawiki-tools-codesniffer already has just such a sniff, we'll copy theirs and modify it to suit our purposes.

This also adjusts .phpcs.xml.dist to ignore more files that aren't actually part of the repo, in preparation for the followup (#17362) that mass-fixes all the new sniff failures this patch introduces.

#### Jetpack product discussion

p9dueE-1Y9-p2 (and some earlier Slack discussion)

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Edit a file to have an unsorted `use` list.
* `composer php:lint file.php` should detect it.
* `composer php:autofix file.php` should fix it.

#### Proposed changelog entry for your changes:

* Developer-only, no changelog entry needed.

### CODE REVIEW

**Comment by anomiex** [comment]
An alternative might be to just pull in [mediawiki/mediawiki-codesniffer](https://packagist.org/packages/mediawiki/mediawiki-codesniffer) via composer and enable just the one sniff, instead of copying the code. But the MediaWiki version uses features that were introduced in PHP 7.1, does that prevent it?

Also if WordPress wants to include something like this in [wp-coding-standards/wpcs](https://packagist.org/packages/wp-coding-standards/wpcs), which IMO they probably should, my guess is they'd want a copy anyway instead of pulling in all MediaWiki's rules to get just the one test.

File: .phpcs.xml.dist [hunk 1]
```diff
@@ -46,7 +46,21 @@
...
+	<exclude-pattern>/docker/wordpress-develop/*</exclude-pattern>
+	<exclude-pattern>/docker/wordpress/*</exclude-pattern>
+
+	<!-- Ignore WordPress included in packages -->
 	<exclude-pattern>/packages/*/wordpress/*</exclude-pattern>
+
+	<!-- Ignore built files -->
+	<exclude-pattern>/_inc/blocks/*</exclude-pattern>
+	<exclude-pattern>/_inc/build/*</exclude-pattern>
+	<exclude-pattern>/_inc/jetpack-strings.php</exclude-pattern>
```
Comment (jeherve) [inline .phpcs.xml.dist:64]: We should be able to remove this one now; we don't use that file anymore.

**Review by jeherve — CHANGES_REQUESTED** [review]
This tests well for me. I am thinking we could merge this into `master`, and then in a follow-up PR, once we've updated all files, elevate the warning into an error?

**Comment by anomiex** [comment]
I'm going to put this on hold for the moment, while we take a more holistic look at MediaWiki sniffs we might want to import in p9dueE-1Y9-p2.

File: .phpcs.xml.dist (hunk 1 above)
Comment (anomiex) [reply .phpcs.xml.dist]: Done, and removed it from .gitignore too.

**Comment by anomiex** [comment]
Doing #17406 instead.

**Pull request closed without merging**